[Issue]: failure to rebuild due to changelog.Debian.gz.

### Problem Description

Doing a basic configure & build failed on a rebuild due to some changelog file:
```
[rocm-core configure] CMake Error at utils.cmake:213 (message):
[rocm-core configure]   Failed to compress: gzip:
[rocm-core configure]   /home/ben/src/TheRock/build/base/rocm-core/build/DEBIAN/changelog.Debian.gz
[rocm-core configure]   already exists; not overwritten
```

The workaround is deleting the file manually and then trying to build again. I got to this point by building once with the default cmake build type and then reconfiguring as Debug before building again. If there are special ways to build a second time it'd be nice to mention them in https://github.com/ROCm/TheRock/tree/main?tab=readme-ov-file#cmake-build-usage, but I suspect this is a bug (missing a `-f` to the gzip command?).

### Operating System

NAME="Ubuntu" VERSION="24.04.2 LTS (Noble Numbat)"

### CPU

 model name      : AMD Ryzen Threadripper PRO 7995WX 96-Cores

### GPU

  Marketing Name:          AMD Radeon Pro W7900 Dual Slot            Name:                    amdgcn-amd-amdhsa--gfx1100                Name:                    amdgcn-amd-amdhsa--gfx11-generic      Name:                    gfx1100                               Marketing Name:          AMD Radeon Pro W7900 Dual Slot            Name:                    amdgcn-amd-amdhsa--gfx1100                Name:                    amdgcn-amd-amdhsa--gfx11-generic   

### ROCm Version

main

### ROCm Component

_No response_

### Steps to Reproduce

_No response_

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information


```sh
ben@noxa-sharkbox:~/src/TheRock$ cmake -B build -GNinja . -DTHEROCK_AMDGPU_FAMILIES=gfx110X-dgpu -DTHEROCK_ENABLE_ALL=OFF -DTHEROCK_ENABLE_CORE=ON -DCMAKE_BUILD_TYPE=Debug
-- Configuring background job pool for 19 concurrent jobs
-- Enabling building tests
-- ROCm version: 7.0.0
-- HIP version: 6.5.0
-- Implicitly enabled features: COMPILER
-- Enabled features:
--   * COMPILER (-DTHEROCK_ENABLE_COMPILER=ON)
--   * CORE_RUNTIME (-DTHEROCK_ENABLE_CORE_RUNTIME=ON)
--   * HIP_RUNTIME (-DTHEROCK_ENABLE_HIP_RUNTIME=ON)
-- Disabled features:
--   * HOST_BLAS (-DTHEROCK_ENABLE_HOST_BLAS=OFF)
--   * HOST_SUITE_SPARSE (-DTHEROCK_ENABLE_HOST_SUITE_SPARSE=OFF)
--   * HIPIFY (-DTHEROCK_ENABLE_HIPIFY=OFF)
--   * ROCPROF_TRACE_DECODER_BINARY (-DTHEROCK_ENABLE_ROCPROF_TRACE_DECODER_BINARY=OFF)
--   * ROCPROFV3 (-DTHEROCK_ENABLE_ROCPROFV3=OFF)
--   * RCCL (-DTHEROCK_ENABLE_RCCL=OFF)
--   * PRIM (-DTHEROCK_ENABLE_PRIM=OFF)
--   * BLAS (-DTHEROCK_ENABLE_BLAS=OFF)
--   * RAND (-DTHEROCK_ENABLE_RAND=OFF)
--   * FFT (-DTHEROCK_ENABLE_FFT=OFF)
--   * SPARSE (-DTHEROCK_ENABLE_SPARSE=OFF)
--   * SOLVER (-DTHEROCK_ENABLE_SOLVER=OFF)
--   * MIOPEN (-DTHEROCK_ENABLE_MIOPEN=OFF)
-- Configured AMDGPU Targets:
--   * gfx1100 : AMD RX 7900 XTX
--   * gfx1101 : AMD RX 7800 XT
--   * gfx1102 : AMD RX 7700S/Framework Laptop 16
--   * Dist bundle: gfx110X-dgpu
-- Building with bundled system dependencies enabled
-- Including subproject therock-boost (from /home/ben/src/TheRock/third-party/boost/cmake_project)
-- Including subproject therock-eigen (from /home/ben/src/TheRock/build/third-party/eigen/source)
-- Including subproject therock-fmt (from /home/ben/src/TheRock/build/third-party/fmt/source)
-- Including subproject therock-googletest (from /home/ben/src/TheRock/build/third-party/googletest/source)
-- Including subproject therock-libdivide (from /home/ben/src/TheRock/build/third-party/libdivide/source)
-- Including subproject therock-msgpack-cxx (from /home/ben/src/TheRock/build/third-party/msgpack-cxx/source)
-- Including subproject therock-nlohmann-json (from /home/ben/src/TheRock/build/third-party/nlohmann-json/source)
-- Including subproject therock-yaml-cpp (from /home/ben/src/TheRock/build/third-party/yaml-cpp/source)
-- Including subproject therock-FunctionalPlus (from /home/ben/src/TheRock/build/third-party/FunctionalPlus/source)
-- Including subproject therock-frugally-deep (from /home/ben/src/TheRock/build/third-party/frugally-deep/source)
-- Including subproject therock-spdlog (from /home/ben/src/TheRock/build/third-party/spdlog/source)
-- Including subproject therock-bzip2 (from /home/ben/src/TheRock/third-party/sysdeps/common/bzip2/.)
-- Including subproject therock-libdrm (from /home/ben/src/TheRock/third-party/sysdeps/linux/libdrm/.)
-- Including subproject therock-numactl (from /home/ben/src/TheRock/third-party/sysdeps/linux/numactl/.)
-- Including subproject therock-sqlite3 (from /home/ben/src/TheRock/third-party/sysdeps/common/sqlite3/.)
-- Including subproject therock-zlib (from /home/ben/src/TheRock/third-party/sysdeps/common/zlib/.)
-- Including subproject therock-zstd (from /home/ben/src/TheRock/third-party/sysdeps/common/zstd/.)
-- Including subproject therock-elfutils (from /home/ben/src/TheRock/third-party/sysdeps/linux/elfutils/.)
-- Including subproject therock-aux-overlay (from /home/ben/src/TheRock/base/aux-overlay)
-- Including subproject rocm-cmake (from /home/ben/src/TheRock/base/rocm-cmake)
-- Including subproject rocm-core (from /home/ben/src/TheRock/base/rocm-core)
-- Including subproject rocm_smi_lib (from /home/ben/src/TheRock/base/rocm_smi_lib)
-- Including subproject rocprofiler-register (from /home/ben/src/TheRock/base/rocprofiler-register)
-- Including subproject rocm-half (from /home/ben/src/TheRock/base/half)
-- Including subproject amd-llvm (from /home/ben/src/TheRock/compiler/amd-llvm/llvm)
-- Including subproject amd-comgr (from /home/ben/src/TheRock/compiler/amd-llvm/amd/comgr)
-- Including subproject hipcc (from /home/ben/src/TheRock/compiler/amd-llvm/amd/hipcc)
-- Including subproject ROCR-Runtime (from /home/ben/src/TheRock/core/ROCR-Runtime)
-- Including subproject rocminfo (from /home/ben/src/TheRock/core/rocminfo)
-- Including subproject hip-clr (from /home/ben/src/TheRock/core/clr)
-- Setting up compile_commands.json export
-- Configuring done (0.6s)
-- Generating done (0.1s)
-- Build files have been written to: /home/ben/src/TheRock/build
ben@noxa-sharkbox:~/src/TheRock$ cmake --build build
[0/2] Re-checking globbed directories...
[16/91] Building sub-project rocm-cmake (in background)
[rocm-cmake] [0/2] Re-checking globbed directories...
[rocm-cmake] ninja: no work to do.
[rocm-cmake completed in 0 seconds]
[18/91] Configure sub-project rocm-core (in background)
FAILED: base/rocm-core/stamp/configure.stamp base/rocm-core/build/CMakeCache.txt base/rocm-core/build/cmake_install.cmake compile_commands_fragment_rocm-core.json /home/ben/src/TheRock/build/base/rocm-core/stamp/configure.stamp /home/ben/src/TheRock/build/base/rocm-core/build/CMakeCache.txt /home/ben/src/TheRock/build/base/rocm-core/build/cmake_install.cmake /home/ben/src/TheRock/build/compile_commands_fragment_rocm-core.json 
cd /home/ben/src/TheRock/build/base/rocm-core/build && /home/ben/.rocm/bin/python3.12 /home/ben/src/TheRock/build_tools/teatime.py --log-timestamps --label rocm-core\ configure --interactive /home/ben/src/TheRock/build/logs/rocm-core_configure.log -- /home/ben/tools/cmake-3.31.7-linux-x86_64/bin/cmake -E env --unset=ROCM_PATH --unset=ROCM_DIR --unset=HIP_PATH --unset=HIP_DIR -- /home/ben/tools/cmake-3.31.7-linux-x86_64/bin/cmake -GNinja -B/home/ben/src/TheRock/build/base/rocm-core/build -S/home/ben/src/TheRock/base/rocm-core -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/ben/src/TheRock/build/base/rocm-core/stage -DTHEROCK_STAGE_INSTALL_ROOT=/home/ben/src/TheRock/build/base/rocm-core/stage -DCMAKE_TOOLCHAIN_FILE=/home/ben/src/TheRock/build/base/rocm-core/_toolchain.cmake -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=/home/ben/src/TheRock/build/base/rocm-core/_init.cmake -DBUILD_SHARED_LIBS=ON -DROCM_VERSION=7.0.0 && /home/ben/tools/cmake-3.31.7-linux-x86_64/bin/cmake -E touch /home/ben/src/TheRock/build/base/rocm-core/build/compile_commands.json && /home/ben/tools/cmake-3.31.7-linux-x86_64/bin/cmake -E touch /home/ben/src/TheRock/build/base/rocm-core/stamp/configure.stamp && /home/ben/tools/cmake-3.31.7-linux-x86_64/bin/cmake -E copy /home/ben/src/TheRock/build/base/rocm-core/build/compile_commands.json /home/ben/src/TheRock/build/compile_commands_fragment_rocm-core.json
[rocm-core configure] -- ROCM_VERSION = 7.0.0
[rocm-core configure] -- VERSION_MAJOR : 7
[rocm-core configure] -- VERSION_MINOR : 0
[rocm-core configure] -- VERSION_PATCH : 0
[rocm-core configure] -- VERSION_POINT : 
[rocm-core configure] -- ROCM_LIBPATCH_VERSION : 70000
[rocm-core configure] -- VERSION_COMMIT_COUNT : 1
[rocm-core configure] -- VERSION_HASH : 8a2b735
[rocm-core configure] -- VERSION_BUILD : 
[rocm-core configure] -- CPACK_DEBIAN_PACKAGE_RELEASE : 
[rocm-core configure] -- CPACK_RPM_PACKAGE_RELEASE : 
[rocm-core configure] -- Using CPACK_PACKAGE_VERSION 7.0.0.70000
[rocm-core configure] -- Using CPACK_DEBIAN_PACKAGE_RELEASE local
[rocm-core configure] -- DEB_PACKAGE_NAME             : rocm-core
[rocm-core configure] -- DEB_PACKAGE_VERSION          : 7.0.0.70000
[rocm-core configure] -- DEB_MAINTAINER_NAME          : ROCm Dev Support
[rocm-core configure] -- DEB_MAINTAINER_EMAIL         : rocm-dev.support@amd.com
[rocm-core configure] -- DEB_COPYRIGHT_YEAR           : 2025
[rocm-core configure] -- DEB_LICENSE                  : MIT
[rocm-core configure] -- DEB_TIMESTAMP                : Wed, 11 Jun 2025 16:45:56 -0700
[rocm-core configure] -- DEB_CHANGELOG_INSTALL_FILENM : changelog.Debian.gz
[rocm-core configure] CMake Error at utils.cmake:213 (message):
[rocm-core configure]   Failed to compress: gzip:
[rocm-core configure]   /home/ben/src/TheRock/build/base/rocm-core/build/DEBIAN/changelog.Debian.gz
[rocm-core configure]   already exists; not overwritten
[rocm-core configure] 
[rocm-core configure] Call Stack (most recent call first):
[rocm-core configure]   CMakeLists.txt:246 (configure_debian_pkg)
[rocm-core configure] 
[rocm-core configure] 
[rocm-core configure] -- Configuring incomplete, errors occurred!
[rocm-core configure completed in 0 seconds]
[26/91] Building sub-project rocprofiler-register (in background)
[rocprofiler-register] [1/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/stacktrace.cc.o
[rocprofiler-register] [2/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/vlog_is_on.cc.o
[rocprofiler-register] [3/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/demangle.cc.o
[rocprofiler-register] [4/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/symbolize.cc.o
[rocprofiler-register] [5/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/flags.cc.o
[rocprofiler-register] [6/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/utilities.cc.o
[rocprofiler-register] [7/20] Building CXX object external/fmt/CMakeFiles/fmt.dir/src/os.cc.o
[rocprofiler-register] [8/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/raw_logging.cc.o
[rocprofiler-register] [9/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/signalhandler.cc.o
[rocprofiler-register] [10/20] Building CXX object external/fmt/CMakeFiles/fmt.dir/src/format.cc.o
[rocprofiler-register] [11/20] Linking CXX static library lib/libfmtd.a
[rocprofiler-register] [12/20] Building CXX object external/glog/CMakeFiles/glog_internal.dir/src/logging.cc.o
[rocprofiler-register] [13/20] Building CXX object external/glog/CMakeFiles/glog.dir/CMakeFiles/glog.cc.o
[rocprofiler-register] [14/20] Linking CXX static library lib/libglogd.a
[rocprofiler-register] [15/20] Building CXX object source/lib/rocprofiler-register/CMakeFiles/rocprofiler-register.dir/details/utility.cpp.o
[rocprofiler-register] [16/20] Building CXX object source/lib/rocprofiler-register/CMakeFiles/rocprofiler-register.dir/details/dl.cpp.o
[rocprofiler-register] [17/20] Building CXX object source/lib/rocprofiler-register/CMakeFiles/rocprofiler-register.dir/details/logging.cpp.o
[rocprofiler-register] [18/20] Building CXX object source/lib/rocprofiler-register/CMakeFiles/rocprofiler-register.dir/rocprofiler_register.cpp.o
[rocprofiler-register] [19/20] Linking CXX shared library lib/librocprofiler-register.so.0.5.0
[rocprofiler-register] [20/20] Creating library symlink lib/librocprofiler-register.so.0 lib/librocprofiler-register.so
[rocprofiler-register completed in 4 seconds]
[27/91] Building sub-project therock-numactl (in background)
ninja: build stopped: subcommand failed.
ben@noxa-sharkbox:~/src/TheRock$ 
```

--- a/patches/amd-mainline/rocm-core/0002-Force-gzip-to-overwrite-an-existing-changelog.patch
+++ b/patches/amd-mainline/rocm-core/0002-Force-gzip-to-overwrite-an-existing-changelog.patch
@@ -1,0 +1,35 @@
+From 159e00067f58fdbe038a6a8d1adc37a3e32d8575 Mon Sep 17 00:00:00 2001
+From: Marius Brehler <marius.brehler@amd.com>
+Date: Wed, 18 Jun 2025 14:10:51 +0000
+Subject: [PATCH 2/2] Force gzip to overwrite an existing changelog
+
+If a compressed changelog exists from a previous build, reconfiguring
+the project fails with
+```
+[rocm-core configure] CMake Error at utils.cmake:213 (message):
+[rocm-core configure]   Failed to compress: gzip:
+[rocm-core configure]   /home/ben/src/TheRock/build/base/rocm-core/build/DEBIAN/changelog.Debian.gz
+[rocm-core configure]   already exists; not overwritten
+```
+
+Add `-f` to force overwriting.
+---
+ utils.cmake | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/utils.cmake b/utils.cmake
+index 42c075b..0a4244b 100644
+--- a/utils.cmake
++++ b/utils.cmake
+@@ -203,7 +203,7 @@ function( configure_debian_pkg PACKAGE_NAME_T COMPONENT_NAME_T PACKAGE_VERSION_T
+       find_program ( DEB_GZIP_EXEC gzip )
+       if(EXISTS "${CMAKE_BINARY_DIR}/DEBIAN/changelog.Debian" )
+         execute_process(
+-          COMMAND ${DEB_GZIP_EXEC} -9 "${CMAKE_BINARY_DIR}/DEBIAN/changelog.Debian"
++          COMMAND ${DEB_GZIP_EXEC} -f -9 "${CMAKE_BINARY_DIR}/DEBIAN/changelog.Debian"
+           WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/DEBIAN"
+           RESULT_VARIABLE result
+           OUTPUT_VARIABLE output
+-- 
+2.43.0
+

